### DISCUSSION TITLE
Cannot escalate to The Hive

### DISCUSSION THREAD

**Discussion by tjbutt58** [Security-Onion-Solutions/securityonion#4323] [Unsupported Versions] [answered]
Whenever I try to escalate to the Hive from the Alerts or Hunt interfaces, I receive a 'Request failed with status code 400' error.
From Kibana the link attempts to access an soctopus page, reports '404 page not found'

I am able to create a test case from the command line using curl using
`curl -XPOST -H 'Authorization: Bearer myapikey' -H 'Content-Type: application/json' http://localhost:9000/thehive/api/case -d '{
"title": "My first case",
"description": "This case has been created by my custom script"
}'`

**Comment by weslambert** [comment]
Are there any clues if you view the logs for Kibana, TheHive, or run docker logs against the so-soctopus container?

**Reply by tjbutt58** [reply]
so-octopus logs are empty
There is nothing apparent in thehive log
There are some messages in kibana.log relating to _ml, eg
...No such index[_ml]...
These are not related by time to my attempts to escalate to the hive.

I can send log fragments if it helps
Where does the ''Alerts' interface send it's logs to?

**Comment by tjbutt58** [comment]
I made a little progress yestaerday by trying to escalate to thehive from Kibana.
A form pops up, filled in with the event details.
When I submit the form, I receive an soctopus error at this URL:
hXXps://192.168.222.102/soctopus/thehive/case/send
> 404 page not found

Soctopus logs show no error.
Possibly an installation error?

**Comment by tjbutt58** [comment]
docker attach <soctopusid> shows

> ko: 400/Request did not complete successfully (401): 401 Unauthorized

When I try this, so it seems the hive_key is probably incorrect, and possibly the cortex_key too.
Can these be re-generated or extracted from The Hive and Cortex?

**Comment by tjbutt58** [comment, marked as answer]
Fixed
I thought to check the api key for the system user against that used in the soctopus docker image.
After some effort, I found the source setting for this key is in /opt/so/saltstack/local/pillar/global.sls
I changed this value to that extracted from TheHive UI, rebooted, escalation now works.

I may need to do the same for Cortex, not sure I can login to it.

**Comment by innovate-support** [comment]
I'm having the same 'status code 400' problem after upgrading to v2.3.70.  When escalating an alert from the Alerts interface I get the red 400 warning.  
![Screenshot from 2021-08-24 08-19-12](https://user-images.githubusercontent.com/20424476/130634149-f859030d-ea3e-4882-bfb0-cdf64d55a2b8.png)

When trying to escalate an alert from Kibana it opens a new tab and then gives a '404 page not found' error.  As previously mentioned the hive and soctopus logs don't seem to reveal clues.

I can log into the Hive web interface manually just fine.  I've checked the previously mentioned user's api key in Hive and it matches the 'hivekey' in global.sls. Is there another key I should be checking?

Also, I ran 'sudo soup' this morning to make sure any fixes were applied.  All went fine with no errors.  I even rebooted for good measure.

**Reply by dougburks** [reply]
@innovate-support Instead of replying to old discussions, please start a new discussion as described at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720. Thanks!
